File: source/ts/src/types/options.ts
export type AyanamsaType =
  | 'lahiri'
  | 'raman'
  | 'krishnamurti'
  | 'true-chitra'
  | 'thirukanitham';
export type Language = 'en' | 'hi';
export type MasaSystem = 'purnimanta' | 'amanta';

/** Default `'whole-sign'`; beyond the polar circles (|φ| > about 66.56°) `'placidus-kp'` can throw
 *  `PanchangError` ('CIRCUMPOLAR' or 'PLACIDUS_DIVERGED'). */
export type HouseSystem = 'whole-sign' | 'equal' | 'placidus-kp';

export interface BirthChartOptions {
  /** Defaults to `'lahiri'`. */
  ayanamsa?: AyanamsaType;
  /** Defaults to `'en'`. */
  language?: Language;
  /** Defaults to `'whole-sign'`. Any other string, `''` included, throws `INVALID_INPUT` (the Go port reads its empty string as the default). */
  houseSystem?: HouseSystem;
  /** Rahu/Ketu node model; default `'mean'` (Meeus polynomial). */
  nodeType?: 'mean' | 'true';
}

/** Default `'all'`; a narrower value keeps that region's festivals plus pan-Indian ones, and `sankranti` always. */
export type FestivalRegion =
  | 'all'
  | 'tamil-nadu'
  | 'kerala'
  | 'karnataka'
  | 'andhra-pradesh'
  | 'telangana'
  | 'west-bengal'
  | 'odisha'
  | 'assam'
  | 'bihar'
  | 'jharkhand'
  | 'gujarat'
  | 'maharashtra'
  | 'goa'
  | 'rajasthan'
  | 'punjab'
  | 'haryana'
  | 'himachal-pradesh'
  | 'uttarakhand'
  | 'uttar-pradesh'
  | 'madhya-pradesh'
  | 'nepal';

/** @deprecated Use the equivalent {@link FestivalRegion} value; `'north-india'` maps to `'all'`. Removal in v6. */
export type LegacyFestivalRegion = 'tamil' | 'bengal' | 'north-india';

export interface InstantPanchangOptions {
  ayanamsa?: AyanamsaType;
  language?: Language;
  /** Default `true`. `false` leaves every anga `endTime` null and each daily anga list at its sunrise entry; the daily `specialYogas` still cover every tithi and nakshatra of the day. */
  computeEndTimes?: boolean;
  /** Default `'purnimanta'` (North Indian). */
  masaSystem?: MasaSystem;
  /** Janma Moon rashi, 0 = Mesha … 11 = Meena. Omit (or pass `null`) to skip `chandraBalam`. */
  janmaRashi?: number;
  /** Janma nakshatra, 0 = Ashwini … 26 = Revati. Omit (or pass `null`) to skip `tarabala`. */
  janmaNakshatra?: number;
  region?: FestivalRegion | LegacyFestivalRegion;
}

/**
 * An optional block of the daily result. A skipped section leaves its own fields at their empty value
 * (`null` or `[]`), with these couplings: `'festivals'` also fills `moon.rise` and `inauspicious.bhadra`,
 * which it needs, and `'eclipse'` adds its grahan entry to `festivals` even when `'festivals'` is off
 * (so `['festivals']` alone lists no grahan).
 */
export type PanchangSection = 'festivals' | 'eclipse' | 'moonTimes' | 'lunarWindows';

export interface PanchangOptions extends InstantPanchangOptions {
  /**
   * UTC offset in minutes (e.g. 330 for IST) or a tz string. Required: untyped JS that omits it gets the host's
   * zone, where the Go port reports INVALID_TIMEZONE. `''` and `'Local'` are not IANA names and throw
   * TIMEZONE_RESOLUTION_FAILED here, while Go's Zone("") reads UTC and Zone("Local") the host's zone.
   */
  timezone: number | string;
  /** Defaults to all sections. A narrowed call's values equal the full call's to the millisecond, except that `festivals` lacks the grahan entry unless `'eclipse'` is requested too. */
  sections?: readonly PanchangSection[];
}

File: source/go/types/options.go
package types

// AyanamsaType selects the sidereal zero point. The ayanamsa is a precession
// offset in degrees subtracted from a tropical longitude to give the
// sidereal one. All five share the IAU precession rate and differ from
// [Lahiri] by a fixed offset. An empty option value resolves to [Lahiri];
// any other unknown value is an ErrInvalidAyanamsa error.
type AyanamsaType string

const (
	// Lahiri is 23.863801 degrees at J2000 plus IAU general precession; the
	// default.
	Lahiri AyanamsaType = "lahiri"
	// Raman is Lahiri minus 1.453010 degrees.
	Raman AyanamsaType = "raman"
	// Krishnamurti is the KP ayanamsa, Lahiri minus 0.079605 degrees.
	Krishnamurti AyanamsaType = "krishnamurti"
	// TrueChitra is True Chitrapaksha, Lahiri minus 0.0006 degrees.
	TrueChitra AyanamsaType = "true-chitra"
	// Thirukanitham is Lahiri plus 0.018456 degrees.
	Thirukanitham AyanamsaType = "thirukanitham"
)

// AllAyanamsaTypes lists the five AyanamsaType values in declaration order:
// Lahiri, Raman, Krishnamurti, TrueChitra, Thirukanitham.
// It is the package's own value, so treat it as read-only: the engine and
// the panchang All functions keep their own copies, so modifying it changes
// no result.
var AllAyanamsaTypes = []AyanamsaType{Lahiri, Raman, Krishnamurti, TrueChitra, Thirukanitham}

// Language selects the language of every user-facing string in a result. An
// empty option value resolves to [LanguageEn], and a value with no
// translation table also renders as English.
type Language string

const (
	// LanguageEn is English, the default.
	LanguageEn Language = "en"
	// LanguageHi is Hindi.
	LanguageHi Language = "hi"
)

// AllLanguages lists LanguageEn then LanguageHi.
// It is the package's own value, so treat it as read-only: the engine and
// the panchang All functions keep their own copies, so modifying it changes
// no result.
var AllLanguages = []Language{LanguageEn, LanguageHi}

// MasaSystem selects the lunar month convention that fills the Index and
// Name of a [ChandraMasaInfo]; both the amanta and the purnimanta index are
// always computed and carried alongside. An empty option value resolves to
// [Purnimanta].
type MasaSystem string

const (
	// Purnimanta ends the masa at the full moon: in Krishna paksha the month
	// is one ahead of the amanta one, except in an adhika masa.
	Purnimanta MasaSystem = "purnimanta"
	// Amanta ends the masa at the new moon: the month index is the sidereal
	// rashi of the Sun at the preceding new moon plus one, so a new moon
	// with the Sun in Meena opens Chaitra.
	Amanta MasaSystem = "amanta"
)

// AllMasaSystems lists Purnimanta then Amanta.
// It is the package's own value, so treat it as read-only: the engine and
// the panchang All functions keep their own copies, so modifying it changes
// no result.
var AllMasaSystems = []MasaSystem{Purnimanta, Amanta}

// FestivalRegion narrows the festivals a result reports. [RegionAll], which
// an empty option value resolves to, applies no filter; any other value
// keeps the festivals tagged for that region plus those tagged for every
// region. The region also picks the civil day a Mesha Sankranti new year
// falls on (Tamil Nadu, Kerala, Punjab, West Bengal, Assam, Odisha). The
// three Legacy values are still accepted and resolve to current ones, see
// [AllLegacyFestivalRegions].
type FestivalRegion string

const (
	// RegionAll applies no regional filter: every festival is reported.
	RegionAll FestivalRegion = "all"
	// RegionTamilNadu selects the Tamil Nadu festival set.
	RegionTamilNadu FestivalRegion = "tamil-nadu"
	// RegionKerala selects the Kerala festival set.
	RegionKerala FestivalRegion = "kerala"
	// RegionKarnataka selects the Karnataka festival set.
	RegionKarnataka FestivalRegion = "karnataka"
	// RegionAndhraPradesh selects the Andhra Pradesh festival set.
	RegionAndhraPradesh FestivalRegion = "andhra-pradesh"
	// RegionTelangana selects the Telangana festival set.
	RegionTelangana FestivalRegion = "telangana"
	// RegionWestBengal selects the West Bengal festival set.
	RegionWestBengal FestivalRegion = "west-bengal"
	// RegionOdisha selects the Odisha festival set.
	RegionOdisha FestivalRegion = "odisha"
	// RegionAssam selects the Assam festival set.
	RegionAssam FestivalRegion = "assam"
	// RegionBihar selects the Bihar festival set.
	RegionBihar FestivalRegion = "bihar"
	// RegionJharkhand selects the Jharkhand festival set.
	RegionJharkhand FestivalRegion = "jharkhand"
	// RegionGujarat selects the Gujarat festival set.
	RegionGujarat FestivalRegion = "gujarat"
	// RegionMaharashtra selects the Maharashtra festival set.
	RegionMaharashtra FestivalRegion = "maharashtra"
	// RegionGoa selects the Goa festival set.
	RegionGoa FestivalRegion = "goa"
	// RegionRajasthan selects the Rajasthan festival set.
	RegionRajasthan FestivalRegion = "rajasthan"
	// RegionPunjab selects the Punjab festival set.
	RegionPunjab FestivalRegion = "punjab"
	// RegionHaryana selects the Haryana festival set.
	RegionHaryana FestivalRegion = "haryana"
	// RegionHimachalPradesh selects the Himachal Pradesh festival set.
	RegionHimachalPradesh FestivalRegion = "himachal-pradesh"
	// RegionUttarakhand selects the Uttarakhand festival set.
	RegionUttarakhand FestivalRegion = "uttarakhand"
	// RegionUttarPradesh selects the Uttar Pradesh festival set.
	RegionUttarPradesh FestivalRegion = "uttar-pradesh"
	// RegionMadhyaPradesh selects the Madhya Pradesh festival set.
	RegionMadhyaPradesh FestivalRegion = "madhya-pradesh"
	// RegionNepal selects the Nepal festival set.
	RegionNepal FestivalRegion = "nepal"
)

// AllFestivalRegions lists every current FestivalRegion in declaration
// order: RegionAll, the twenty Indian states from RegionTamilNadu to
// RegionMadhyaPradesh, then RegionNepal. The deprecated aliases in
// AllLegacyFestivalRegions are not included.
// It is the package's own value, so treat it as read-only: the engine and
// the panchang All functions keep their own copies, so modifying it changes
// no result.
var AllFestivalRegions = []FestivalRegion{
	RegionAll,
	RegionTamilNadu, RegionKerala, RegionKarnataka, RegionAndhraPradesh, RegionTelangana,
	RegionWestBengal, RegionOdisha, RegionAssam, RegionBihar, RegionJharkhand,
	RegionGujarat, RegionMaharashtra, RegionGoa, RegionRajasthan,
	RegionPunjab, RegionHaryana, RegionHimachalPradesh, RegionUttarakhand,
	RegionUttarPradesh, RegionMadhyaPradesh,
	RegionNepal,
}

const (
	// LegacyRegionTamil is deprecated; it resolves to RegionTamilNadu.
	LegacyRegionTamil FestivalRegion = "tamil"
	// LegacyRegionBengal is deprecated; it resolves to RegionWestBengal.
	LegacyRegionBengal FestivalRegion = "bengal"
	// LegacyRegionNorthIndia is deprecated; it resolves to RegionAll.
	LegacyRegionNorthIndia FestivalRegion = "north-india"
)

// AllLegacyFestivalRegions lists the three deprecated FestivalRegion values,
// LegacyRegionTamil, LegacyRegionBengal and LegacyRegionNorthIndia. Each is
// accepted as input and resolved to its current value, with a one-time
// deprecation warning when a warner is configured.
// It is the package's own value, so treat it as read-only: the engine and
// the panchang All functions keep their own copies, so modifying it changes
// no result.
var AllLegacyFestivalRegions = []FestivalRegion{
	LegacyRegionTamil, LegacyRegionBengal, LegacyRegionNorthIndia,
}
